workspaces/adventure-pack/goodies/python3/src/traverse_preorder/__init__.py

```diff
@@ -0,0 +1,25 @@
+from collections import deque
+from typing import Generator, Optional
+
+
+class TreeNode:
+    def __init__(
+        self,
+        val: int = 0,
+        left: Optional["TreeNode"] = None,
+        right: Optional["TreeNode"] = None,
+    ):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def traverse_preorder(root: Optional[TreeNode]) -> Generator[int, None, None]:
+    stack = [root]
+    while stack:
+        node = stack.pop()
+
+        if node:
```

Comment:
I personally have a slight preference for the early return/continue style (i.e. `if not node: continue`) but I'm willing to be convinced otherwise if you have a stronger preference for this style.

Reply:
Was trying to maintain the same style as in Typescript. I will probably modify as not to add `None` values.